agent/xds/proxystateconverter/clusters.go

```diff
@@ -462,7 +462,7 @@ func (s *Converter) makeAppCluster(cfgSnap *proxycfg.ConfigSnapshot, name, pathP
 	if protocol == "" {
 		protocol = cfg.Protocol
 	}
-	namedCluster.cluster.Protocol = protocol
+	namedCluster.cluster.Protocol = protocolMap[protocol]
```

Comment:
proxystate cluster.Protocol is now an enumeration.  so we need to convert the string from config entry to the protocol enum.